src/Symfony/Bundle/FrameworkBundle/Resources/config/secrets.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://symfony.com/schema/dic/services https://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="secrets.vault" class="Symfony\Bundle\FrameworkBundle\Secrets\SodiumVault">
+            <argument>%kernel.project_dir%/config/secrets/%kernel.environment%</argument>
+            <argument type="service" id="secrets.decryption_key" on-invalid="ignore" />
+        </service>
+
+        <!--
+            LazyString::fromCallable() is used as a wrapper to lazily read the SYMFONY_DECRYPTION_SECRET var from the env.
+            By overriding this service and using the same strategy, the decryption key can be fetched lazily from any other service if needed.
+        -->
+        <service id="secrets.decryption_key" class="object">
+            <factory class="Symfony\Component\DependencyInjection\LazyString" method="fromCallable" />
+            <argument type="service">
+                <service class="Closure">
+                    <factory class="Closure" method="fromCallable" />
+                    <argument type="collection">
+                        <argument type="service" id="service_container" />
+                        <argument>getEnv</argument>
+                    </argument>
+                </service>
+            </argument>
+            <argument>base64:default::SYMFONY_DECRYPTION_SECRET</argument>
```

Comment:
The name of this variable conflict with the `_SECRET` suffix: Which has 2 impacts:
- DotenvVault display in stdout when runing the command `secrets:list -r`
- because of urlencode/decode this variable can not be really used as secret DotenvVault

tips: When storing the private key into the DotenvVault (and using it as a secret) (`php -r 'echo base64_encode(require "config/secrets/dev/dev.sodium.decrypt.key");'|./bin/console s:s SYMFONY_DECRYPTION --local -`) the variable is url encoded and config should be fixed with 
```
framework:
 secrets:
  decryption_env_var: 'base64:default::secret:SYMFONY_DECRYPTION'
```

Reply:
> because of urlencode/decode this variable can not be really used as secret DotenvVault

urlencoding now replaced by shell escaping: no issue anymore. see #34023 for LFs